src/enqueue_links.js

```diff
@@ -0,0 +1,214 @@
+import _ from 'underscore';
+import log from 'apify-shared/log';
+import { checkParamOrThrow } from 'apify-client/build/utils';
+import { checkParamPrototypeOrThrow } from 'apify-shared/utilities';
+import { RequestQueue, RequestQueueLocal } from './request_queue';
+import Request from './request';
+import PseudoUrl from './pseudo_url';
+
+/**
+ * To enable direct use of the Actor UI `pseudoUrls` output while keeping high performance,
+ * all the pseudoUrls from the output are only constructed once and kept in a cache
+ * by the `enqueueLinks()` function.
+ * @ignore
+ */
+const enqueueLinksCache = new Map();
+export const MAX_ENQUEUE_LINKS_CACHE_SIZE = 1000;
```

Comment:
FYI we have an `LruCache` object exactly for this use case.

Reply:
It only supports `string` keys unfortunately.